server/src/test/java/io/envoyproxy/controlplane/server/V3DiscoveryServerAdsAllowDefaultEmptyEdsIT.java

```diff
@@ -0,0 +1,131 @@
+package io.envoyproxy.controlplane.server;
+
+import static io.envoyproxy.envoy.config.core.v3.ApiVersion.V3;
+import static io.restassured.RestAssured.given;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.containsString;
+
+import io.envoyproxy.controlplane.cache.TestResources;
+import io.envoyproxy.controlplane.cache.v3.SimpleCache;
+import io.envoyproxy.controlplane.cache.v3.Snapshot;
+import io.envoyproxy.controlplane.server.serializer.DefaultProtoResourcesSerializer;
+import io.envoyproxy.envoy.config.cluster.v3.Cluster;
+import io.envoyproxy.envoy.config.endpoint.v3.ClusterLoadAssignment;
+import io.envoyproxy.envoy.config.listener.v3.Listener;
+import io.envoyproxy.envoy.config.route.v3.RouteConfiguration;
+import io.grpc.netty.NettyServerBuilder;
+import io.restassured.http.ContentType;
+import java.util.Collections;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.testcontainers.containers.Network;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableList;
+
+/**
+ * This test verifies when allowDefaultEmptyEdsUpdate is true, we whether envoy control
+ * will send EDS response when some clusters are missing in ClusterLoadAssignments in
+ * the snapshot.
+ * When allowDefaultEmptyEdsUpdate is false, EDS response will not be sent from envoy-control
```

Comment:
not easy to write test case for this scenario. I did spend quite amount of time. 

For example, 

1. if I check warming cluster, how do you know warming cluster is not because we did not wait long enough. 
2. I also create latches for each typeUrl, I want to check cluster callback is triggered while endpoint callback is not. Test never works reliably. 

but I did verify allowDefaultEmptyEdsUpdate as false path manually